0-base-protocol-background.md

```diff
@@ -0,0 +1,39 @@
+2. Data flows through an external transport layer (for inbound messages from a counterparty) or JSON rpc interface (for user-initiated inputs), gets validated by the protocol against previous state, and then gets stored in store.
+3. The output of state transitions is a set of commitments that can be put onchain in the event of a dispute.
+
+It is different in the following ways:
+1. The "application" lifecycle is named `create`--> `join`--> `update`--> `close`. The same cycle is used both for "subchannels"/virtual channels (i.e. apps), **and** for the ledger channel. This dramatically simplifies the interfaces.
+2. The core protocol handles deposit/withdraw natively. //TODO How do channel topups and partial withdraws work?
+3. There are no more multisig contracts. Instead there's a global `assetHolder` contract per asset. //TODO How do we deposit directly to an address? How do we allow counterparty to pay for withdrawal?
+4. The StateChannels wallet does not handle message dispatching and responding directly. Instead, it uses a `pushMessage` pattern whereby the implementer pushes updates into the wallet. This dramatically simplifies adding middleware for encryption, message retries, caching, etc.
+5. The protocol extends turn based updating from being only an "app"-level feature to instead be throughout the protocol. This removes the need for distributed locks, dramatically reduces the risk of states getting out of sync with each other (TODO: verify), and allows parties to receive updates entirely out of order with no problems.
+6. The application solidity code no longer defines a state machine but instead a simpler interface which just ensures that a given transition is valid. This means that calculating state transitions themselves is outside the scope of the wallet.
+7. The wallet is more tightly coupled with the store implementation and cannot easily be abstracted into a general interface with pluggable stores.
+8. For now, the wallet does not allow for an externally passed in signer. This is something that can be changed, however.
+
+## Terminology
+Some terms and naming conventions have been changed in the new protocol:
+- **Sub/virtual/ledger channel**: A ledger channel refers to a channel whose funding is in an onchain address. A "sub" channel is what we currently call an "app". A virtual channel is a sub channel constructed over multiple ledger channels through one or many intermediaries.
+- Distinction between apps/channels: In StateChannels, there is no difference between ledger vs sub/virtual channels. This means `createChannel` refers to *both* creating a channel onchain and to installing an app.
+- **ChannelId**: Corresponds to existing `multisigAddress` for a ledger channnel or existing `appIdentityHash` for a sub/virtual channel.
+- **ParticipantId**: Unique identifier for a given participant. For now, this is the equivalent of `signerAddress`.
```

Comment:
In my understanding, this is actually equivalent to the `publicIdentifier`. It is used to route messages, and the `Participant` type includes information about the signing address and destination address (where funds are sent in the case of a dispute)

Reply:
StateChannels doesn't actually route the messages, so publicIdentifier is entirely on our end. 

```
getFirstParticipant: async function(): Promise<Participant> {
  const signingAddress = await Store.getOrCreateSigningAddress();
  return {participantId: signingAddress, signingAddress, destination: makeDestination(HashZero)};
}
```

As per the current implementation, signer address and participantId are the same.